src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -580,24 +645,27 @@ private void SetPatternLocalScopes(BoundObjectPattern pattern)
             return null;
         }
 
-        private PlaceholderRegion GetArgumentPlaceholders(BoundExpression? receiverOpt, ImmutableArray<BoundExpression> arguments)
+        private void VisitArgumentsAndGetArgumentPlaceholders(BoundExpression? receiverOpt, ImmutableArray<BoundExpression> arguments)
         {
-            var placeholders = ArrayBuilder<(BoundValuePlaceholderBase, uint)>.GetInstance();
-            foreach (var arg in arguments)
+            Visit(receiverOpt);
+
+            for (int i = 0; i < arguments.Length; i++)
             {
+                var arg = arguments[i];
                 if (arg is BoundConversion { ConversionKind: ConversionKind.InterpolatedStringHandler, Operand: BoundInterpolatedString or BoundBinaryOperator } conversion)
                 {
                     var interpolationData = conversion.Operand.GetInterpolatedStringHandlerData();
-                    GetInterpolatedStringPlaceholders(placeholders, interpolationData, receiverOpt, arguments);
+                    var placeholders = ArrayBuilder<(BoundValuePlaceholderBase, uint)>.GetInstance();
+                    GetInterpolatedStringPlaceholders(placeholders, interpolationData, receiverOpt, i, arguments);
+                    _ = new PlaceholderRegion(this, placeholders);
```

Comment:
>_ = new PlaceholderRegion(this, placeholders); [](http://example.com/codeflow?start=20&length=46)

Are we relying on side-effects of this constructor? Consider extracting a reusable helper with meaningful name and using it instead.